integrations/ConvertFlow/utils.js

```diff
@@ -0,0 +1,83 @@
+import {
+  getHashFromArray,
+  isDefinedAndNotNullAndNotEmpty,
+} from "../utils/commonUtils";
+
+// default mapping for the events
+const standardEventsListMapping = {
+  cfReady: "CTA Ready",
+  cfView: "CTA Viewed",
+  cfConversion: "CTA Converted",
+  cfCompletion: "CTA Completed",
+  cfSubmit: "CTA Form Submitted",
+  cfAddToCart: "Product Added to Cart",
+  cfClosed: "CTA Closed",
+};
+
+/**
+ * This function is used to trigger a callback.
+ * @param {*} standardEventsMap - mapping of events done by the user
+ * @param {*} eventName - standard event name
+ * @param {*} data
+ */
+const makeACall = (standardEventsMap, eventName, data) => {
+  // stroring all the supported standard event names
+  const eventNames = Object.keys(standardEventsMap);
+  // stroing default event name in the updatedEvent
+  let updatedEvent = standardEventsListMapping[eventName];
+  // Updating the event name with any mapping from the webapp
+  eventNames.forEach((event) => {
+    if (standardEventsMap[event] === eventName) {
+      updatedEvent = event;
+    }
+  });
+  // Populating Properties
+  const properties = {};
+  if (data) {
+    if (data.cta) {
+      const { cta } = data;
+      properties.cta_name = cta.name;
+      properties.cta_type = cta.cta_type;
+      properties.cta_id = cta.id;
+    }
+    if (data.variant) {
+      properties.cta_variant = data.variant;
+    }
+    if (data.step) {
+      properties.cta_step = data.step;
+    }
+  }
```

Comment:
Can be moved to a function called `populatingProperties`

Reply:
done

Reply:
done